### PR TITLE
[JENKINS-70392] RealJenkinsRule testing

### PR DESCRIPTION
**Pull request by Vlatombe**
Re-enable tests depending on RealJenkinsRule.

* [x] Downstream of https://github.com/jenkinsci/jenkins-test-harness/pull/541

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,6 +49,7 @@
     <no-test-jar>false</no-test-jar>
     <useBeta>true</useBeta>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
+    <jenkins-test-harness.version>1919.v8b_ea_09f5a_b_e2</jenkins-test-harness.version>
```
Comment (jglick) [inline pom.xml:52]:
```suggestion
    <jenkins-test-harness.version>1919.v8b_ea_09f5a_b_e2</jenkins-test-harness.version> <!-- TODO https://github.com/jenkinsci/jenkins-test-harness/pull/541 -->
```

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/AbstractKubernetesPipelineRJRTest.java [hunk 2]
```diff
@@ -18,7 +18,16 @@ public abstract class AbstractKubernetesPipelineRJRTest {
     public TestName name = new TestName();
 
     @Rule
-    public RealJenkinsRule rjr = new RealJenkinsRule();
+    public RealJenkinsRule rjr;
+    {
+        rjr = new RealJenkinsRule();
+        String port = System.getProperty("port");
```
Comment (jglick) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/AbstractKubernetesPipelineRJRTest.java:24]: Where would this get set? https://github.com/jenkinsci/kubernetes-plugin/blob/87f08371da37a00e3be16c6b039699de06c2959d/test-in-k8s.sh#L33 is set in the Maven JVM, not passed along to the Surefire booter currently: https://github.com/jenkinsci/kubernetes-plugin/blob/87f08371da37a00e3be16c6b039699de06c2959d/pom.xml#L268-L276

**Comment by jglick** [comment]
Why not just define a default

```xml
<port>0</port>
```
?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/WithContextClassLoader.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package org.csanchez.jenkins.plugins.kubernetes;
+
+// TODO post 2.362 use jenkins.util.SetContextClassLoader
+class WithContextClassLoader implements AutoCloseable {
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/WithContextClassLoader.java:4]: Needs to merge in trunk I think?

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/AbstractKubernetesPipelineRJRTest.java (hunk 2 above)
Comment (jglick) [reply src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/AbstractKubernetesPipelineRJRTest.java:30]: For reference, did you figure this out? I see https://ci.jenkins.io/job/Plugins/job/kubernetes-plugin/job/PR-1290/6/testReport/org.csanchez.jenkins.plugins.kubernetes.pipeline/KubernetesPipelineRJRTest/kind___basicPipeline/ passing with

```
Overriding port using system property: 7221
```

but it is unclear to me how it works.

**Review by jglick — APPROVED** [review]

**Comment by Vlatombe** [comment]
Looks like maven passes down its command line args down to surefire launcher

```
[DEBUG] sun.java.command: org.codehaus.plexus.classworlds.launcher.Launcher test -Dtest=org.csanchez.jenkins.plugins.kubernetes.pipeline.KubernetesPipelineRJRTest#basicPipeline -Dport=8888 -X
```

**Comment by Vlatombe** [comment]
`mvn test -Dtest=org.csanchez.jenkins.plugins.kubernetes.pipeline.KubernetesPipelineRJRTest#basicPipeline -Dport=8888 -X`

Looks like surefire passes down the maven option to surefire system properties.
```
[DEBUG] Setting system property [port]=[8888]
```

File: pom.xml [hunk 4]
```diff
@@ -49,6 +49,7 @@
     <no-test-jar>false</no-test-jar>
     <useBeta>true</useBeta>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
+    <jenkins-test-harness.version>1921.v1f1c29de9da_2</jenkins-test-harness.version> <!-- TODO https://github.com/jenkinsci/jenkins-test-harness/pull/541 -->
```
Comment (Vlatombe) [inline pom.xml:52]:
```suggestion
    <jenkins-test-harness.version>1922.v91779ee45b_5e</jenkins-test-harness.version>
```

**Pull request merged**